### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by xingwangsfu** [kpzhang93/MTCNN_face_detection_alignment#1]
Hi, 

Thanks for sharing the code. I want to reproduce the result and maybe make some modifications on the network architecture.Is it possible to share the training code with us? especially, how to prepare the dataset. Thanks.

**Comment by kpzhang93** [comment]
The training code may be released few months later (I am too busy to sort out them). I think you can prepare the dataset by yourself. The main idea can be found in our paper and you can e-mail to me for more details.

**Comment by anjiang2016** [comment]
few months!  too long to wait,come on guys

**Comment by oppo62258801** [comment]
Come on guys! There have been about 5 months since you said sharing the training code and dataset. Ｉ think your training code and the way to prepare the dataset would help us a lot. I hope you can give us a hand and we would thank you so much.

**Comment by amds450076077** [comment]
How to prepare the dataset and can you give me a hand?
How to set the weight in the different loss layers?
Is the dataset of the PNet,RNet and ONet different?
How to randomly crop the wider_face images and how many faces in the one cropped image?

**Comment by CongWeilin** [comment]
@amds450076077 My project provide training data [https://github.com/CongWeilin/mtcnn-caffe](https://github.com/CongWeilin/mtcnn-caffe)

**Comment by jianbaba** [comment]
哥，给您发邮件请教问题了，但一直没收到回复。求训练代码。 @kpzhang93

**Comment by whcjb** [comment]
Where is the base trust among people, it's one YEAR passed, but where is the training code???

**Comment by whcjb** [comment]
**You've broken your promise to the public.**

**Comment by Hzzone** [comment]
@whcjb 你要不要这么逗。。。

**Comment by jerryhouuu** [comment]
Looking forward to the release of Training code !

**Comment by whcjb** [comment]
This work is magic, when I use the method in the paper, the angle problem cant be solved, i.e. the skew faces have no solution. Planning to use spatial transform network. Any ideas about the skew faces? thanks

**Comment by zHanami** [comment]
hello,i would like to know the speed of detect,in paper it just said 99fps ,but my speed is juast 2fps,what about you? my gpu is tatian x

**Comment by Dawson-huang** [comment]
@CongWeilin Hi, I reproduce your project, get this question:  cls_loss = 0.693147

I0222 15:35:19.157838  2982 sgd_solver.cpp:105] Iteration 60000, lr = 6.4e-08
I0222 15:35:33.670897  2982 solver.cpp:218] Iteration 61000 (68.902 iter/s, 14.5134s/1000 iters), loss = 0.795055
I0222 15:35:33.671394  2982 solver.cpp:237]     Train net output #0: cls_Acc = 0
I0222 15:35:33.671587  2982 solver.cpp:237]     Train net output #1: cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0222 15:35:33.671739  2982 solver.cpp:237]     Train net output #2: pts_loss = 0.203815 (* 0.5 = 0.101908 loss)
I0222 15:35:33.671883  2982 solver.cpp:237]     Train net output #3: roi_loss = 0 (* 0.5 = 0 loss)
I0222 15:35:33.671980  2982 sgd_solver.cpp:105] Iteration 61000, lr = 6.4e-08
I0222 15:35:47.946030  2982 solver.cpp:218] Iteration 62000 (70.0535 iter/s, 14.2748s/1000 iters), loss = 0.786254
I0222 15:35:47.946133  2982 solver.cpp:237]     Train net output #0: cls_Acc = 0
I0222 15:35:47.946175  2982 solver.cpp:237]     Train net output #1: cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0222 15:35:47.946233  2982 solver.cpp:237]     Train net output #2: pts_loss = 0.186213 (* 0.5 = 0.0931064 loss)
I0222 15:35:47.946262  2982 solver.cpp:237]     Train net output #3: roi_loss = 0 (* 0.5 = 0 loss)
I0222 15:35:47.946286  2982 sgd_solver.cpp:105] Iteration 62000, lr = 6.4e-08

Do you know the reason?

**Comment by ouguozhen** [comment]
@Dawson-huang do you solve this problem? i meet the same

**Comment by Fei-dong** [comment]
@Dawson-huang I also meet this problem,you solve it??? thx

**Comment by Fei-dong** [comment]
@CongWeilin
I0222 15:35:19.157838 2982 sgd_solver.cpp:105] Iteration 60000, lr = 6.4e-08
I0222 15:35:33.670897 2982 solver.cpp:218] Iteration 61000 (68.902 iter/s, 14.5134s/1000 iters), loss = 0.795055
I0222 15:35:33.671394 2982 solver.cpp:237] Train net output #0: cls_Acc = 0
I0222 15:35:33.671587 2982 solver.cpp:237] Train net output #1: cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0222 15:35:33.671739 2982 solver.cpp:237] Train net output #2: pts_loss = 0.203815 (* 0.5 = 0.101908 loss)
I0222 15:35:33.671883 2982 solver.cpp:237] Train net output #3: roi_loss = 0 (* 0.5 = 0 loss)
I0222 15:35:33.671980 2982 sgd_solver.cpp:105] Iteration 61000, lr = 6.4e-08
I0222 15:35:47.946030 2982 solver.cpp:218] Iteration 62000 (70.0535 iter/s, 14.2748s/1000 iters), loss = 0.786254
I0222 15:35:47.946133 2982 solver.cpp:237] Train net output #0: cls_Acc = 0
I0222 15:35:47.946175 2982 solver.cpp:237] Train net output #1: cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0222 15:35:47.946233 2982 solver.cpp:237] Train net output #2: pts_loss = 0.186213 (* 0.5 = 0.0931064 loss)
I0222 15:35:47.946262 2982 solver.cpp:237] Train net output #3: roi_loss = 0 (* 0.5 = 0 loss)
I0222 15:35:47.946286 2982 sgd_solver.cpp:105] Iteration 62000, lr = 6.4e-08

**Comment by Fei-dong** [comment]
@CongWeilin 
I0222 15:35:19.157838 2982 sgd_solver.cpp:105] Iteration 60000, lr = 6.4e-08
I0222 15:35:33.670897 2982 solver.cpp:218] Iteration 61000 (68.902 iter/s, 14.5134s/1000 iters), loss = 0.795055
I0222 15:35:33.671394 2982 solver.cpp:237] Train net output #0: cls_Acc = 0
I0222 15:35:33.671587 2982 solver.cpp:237] Train net output #1: cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0222 15:35:33.671883 2982 solver.cpp:237] Train net output #3: roi_loss = 0 (* 0.5 = 0 loss)
I0222 15:35:33.671980 2982 sgd_solver.cpp:105] Iteration 61000, lr = 6.4e-08
I0222 15:35:47.946030 2982 solver.cpp:218] Iteration 62000 (70.0535 iter/s, 14.2748s/1000 iters), loss = 0.786254
I0222 15:35:47.946133 2982 solver.cpp:237] Train net output #0: cls_Acc = 0
I0222 15:35:47.946175 2982 solver.cpp:237] Train net output #1: cls_loss = 0.693147 (* 1 = 0.693147 loss)
I0222 15:35:47.946262 2982 solver.cpp:237] Train net output #3: roi_loss = 0 (* 0.5 = 0 loss)
I0222 15:35:47.946286 2982 sgd_solver.cpp:105] Iteration 62000, lr = 6.4e-08

分开训练是收敛的！

**Comment by lzwhard** [comment]
@Fei-dong
@Dawson-huang 
Hi guys, I also encounter this problem, will we can discuss together? my qq 739544972.

@CongWeilin 
Thanks for your work, and could you open the issue?
